theme.inc

```diff
@@ -95,3 +109,209 @@ function _gent_base_get_timeline_display(Paragraph $entity) {
   $isProgram = $entity->get('field_timeline_as_program')->value;
   return ($isProgram === '1') ? 'alt' : 'normal';
 }
+
+/**
+ * Count the number of input fields in a render array.
+ *
+ * @param array $array
+ *   Render array to loop over.
+ *
+ * @return int
+ *   The number of input fields counted.
+ */
+function _gent_base_count_input_fields(array $array) {
+  $input_count = 0;
+  $fields_types = [
+    'checkbox',
+    'checkboxes',
+    'date',
+    'email',
+    'file',
+    'number',
+    'password',
+    'radio',
+    'radios',
+    'select',
+    'textarea',
+    'textfield',
+    'url',
+    'webform_address',
+  ];
+
+  foreach ($array as &$field) {
+    if (!is_array($field) || !isset($field['#type'])) {
+      continue;
+    }
+
+    if (!in_array($field['#type'], $fields_types)) {
+      continue;
+    }
+
+    $input_count++;
+  }
+
+  return $input_count;
+}
+
+/**
+ * Adds the step title to the step for webforms with multiple pages.
+ *
+ * @param array $form
+ *   Render array for the form.
+ */
+function _gent_base_add_webform_step_title(array &$form) {
+  foreach (Element::children($form['elements']) as $key) {
+    $step = &$form['elements'][$key];
+
+    if ($step['#type'] !== 'container') {
+      continue;
+    }
+
+    // Generate the markup.
+    $html = '<h2>' . $step['#title'] . '</h2>';
+
+    if (_gent_base_count_input_fields($step) > 0) {
+      $html .= '<p class="form-disclaimer">' . t('Fill in all fields unless otherwise stated') . '</p>';
+    }
+
+    array_unshift($step, [
+      '#markup' => $html,
+      '#parents' => [],
+    ]);
+  }
+}
+
+/**
+ * Loop webform elements and add theming parameters for the desired elements.
+ *
+ * @param array $elements
+ *   Build array for the elements that have to be altered.
+ */
+function _gent_base_loop_webform_elements(array &$elements) {
+  $children = Element::children($elements);
+
+  if ($children === ['elements']) {
+    _gent_base_loop_webform_elements($elements['elements']);
+    return;
+  }
+
+  foreach ($children as $key) {
+    $element = &$elements[$key];
+    if (!is_array($element) || !isset($element['#type'])) {
+      continue;
+    }
+
+    _gent_base_loop_webform_elements_options($element);
+    _gent_base_loop_webform_elements_options_other($element);
+    _gent_base_loop_webform_elements_email_confirm($element);
+    _gent_base_loop_webform_elements_datetime($element);
+    _gent_base_loop_webform_elements_select_other($element);
+    _gent_base_loop_webform_group($element);
+  }
+}
+
+/**
+ * Add template variables for option elements.
+ *
+ * @param array $element
+ *   Build array a form element.
+ */
+function _gent_base_loop_webform_elements_options(array &$element) {
+  $optionTypes = ['checkboxes', 'radios'];
+  if (!isset($element['#type']) || !in_array($element['#type'], $optionTypes)) {
```

Comment:
Move the check if the `!isset($element['#type'])` is set to the loop (as it is repeated in every function).